src/serializer/ResidualHeapSerializer.js

```diff
@@ -1229,6 +1229,24 @@ export class ResidualHeapSerializer {
     return serializedValue;
   }
 
+  _serializeAbstractValue(val: AbstractValue): void | BabelNodeExpression {
+    invariant(val.kind !== "sentinel member expression", "invariant established by visitor");
+    if (val.hasIdentifier()) {
+      return this._serializeAbstractValueHelper(val);
+    } else {
+      // This abstract value's dependencies may have been declared
+      // but still need to check it again in case their serialized bodies are in different generator scope.
+      this.emitter.emitNowOrAfterWaitingForDependencies(val.args, () => {
+        const serializedValue = this._serializeAbstractValueHelper(val);
+        // Fetch the previous assigned identifier by residualHeapValueIdentifiers.setIdentifier()
+        // before _serializeAbstractValue() is called.
+        let uid = this.residualHeapValueIdentifiers.getIdentifierAndIncrementReferenceCount(val);
+        let declar = t.variableDeclaration("var", [t.variableDeclarator(uid, serializedValue)]);
```

Comment:
I still don't understand how it can be that we don't have two competing (or redundant) variable declarations here.

Reply:
Reading through the code in serializeValue, I start to be convinced that this is okay...